crates/fuel-core/src/graphql_api/metrics_extension.rs

```diff
@@ -118,4 +121,14 @@ impl Extension for MetricsExtInner {
 
         res
     }
+
+    async fn validation(
```

Comment:
It doesn't look like this gets used by anything in this PR... where is it used?

Reply:
I'm not very familiar with async-graphql but the trait seems to have a default implementation and I override it. For me this is call by async-graphql itself when triggering validation step of the query

Reply:
It is how `async-graphql` library works. It has a bunch of default methods that are called during the processing of the request. Valuation is one of the phases there.
<img width="890" alt="image" src="https://github.com/user-attachments/assets/a7a2ade5-3c96-42b0-82c1-d49fc361b464">

Reply:
Ah. I missed it was a trait impl. That makes sense now.